docs/proposals/playbook_consolidation.md

```diff
@@ -0,0 +1,183 @@
+  top of `playbooks/init/main.yml`
+* All other related files for initialization have been moved to `playbooks/init`
+
+The `initialize_host_groups.yml` playbook is only one play with one task for
+importing variables for inventory group conversions.  This task could be further
+consolidated with the play in `evaluate_groups.yml`.
+
+The new standard initialization playbook would be
+`playbooks/init/main.yml`.
+
+
+```
+ 
+> $ tree openshift-ansible/playbooks/init
+.
+├── evaluate_groups.yml
+├── initialize_facts.yml
+├── initialize_host_groups.yml
+├── initialize_openshift_repos.yml
+├── initialize_openshift_version.yml
+├── main.yml
+├── roles -> ../../roles
+├── validate_hostnames.yml
+└── vars
+    └── cluster_hosts.yml
+```
+
+```yaml
+# openshift-ansible/playbooks/init/main.yml
+---
+- include: initialize_host_groups.yml
+
+- include: evaluate_groups.yml
+
+- include: initialize_facts.yml
+
+- include: validate_hostnames.yml
+
+- include: initialize_openshift_repos.yml
+
+- include: initialize_openshift_version.yml
+```
+
+### `byo` and `common` Playbook Consolidation
+Historically, the `byo` directory coexisted with other platform directories
+which contained playbooks that then called into `common` playbooks to perform
+common installation steps for all platforms.  Since the other platform
+directories have been removed this separation is no longer necessary.
+
+In the example below, the following changes have been made:
+* `playbooks/byo/openshift-master` renamed to
+  `playbooks/openshift-master`
+* `playbooks/common/openshift-master` renamed to
+  `playbooks/openshift-master/private`
+* Original `byo` entry point playbooks have been updated to include their
+  respective playbooks from `private/`.
+* Symbolic links have been updated as necessary
+
+All user consumable playbooks are in the root of `openshift-master` and no entry
+point playbooks exist in the `private` directory.  Maintaining the separation
+between entry point playbooks and the private playbooks allows individual pieces
+of the deployments to be used as needed by other components.
+
+```
+openshift-ansible/playbooks/openshift-master 
+> $ tree
+.
+├── config.yml
+├── private
+│   ├── additional_config.yml
+│   ├── config.yml
+│   ├── filter_plugins -> ../../../filter_plugins
+│   ├── library -> ../../../library
+│   ├── lookup_plugins -> ../../../lookup_plugins
+│   ├── restart_hosts.yml
+│   ├── restart_services.yml
+│   ├── restart.yml
+│   ├── roles -> ../../../roles
+│   ├── scaleup.yml
+│   └── validate_restart.yml
+├── restart.yml
+└── scaleup.yml
+```
+
+```yaml
+# openshift-ansible/playbooks/openshift-master/config.yml
+---
+- include: ../init/main.yml
+
+- include: private/config.yml
+```
+
+The following examples shows how multiple components would be combined to
+perform a complete install.
+
+```yaml
+# openshift-ansible/playbooks/openshift-cluster/config.yml
+---
+- include: ../init/main.yml
+
+- include: private/config.yml
```

Comment:
I don't see a purpose for this extra level of redirection.  Why not put the includes directly in this file?

Reply:
`private/config.yml` is included by other things and therefore needs to be separate.  Such as being called from `openshift-master/scaleup.yml` and `openshift-cluster/config.yml`.  The initialization step should not be called each time the individual component is called.

Reply:
I must have had a misunderstanding of your question and I might have confused two different playbooks.  Could you restate what you are asking or are requesting?

Reply:
@michaelgugino I've pushed a new commit with changes to the openshift-cluster section.  Please let me know if this represents what you are suggesting.